src/main/java/com/github/charithe/kafka/KafkaJunitRule.java

```diff
@@ -84,8 +90,12 @@ public KafkaJunitRule(final int kafkaPort) {
 
     @Override
     protected void before() throws Throwable {
-        zookeeper = new TestingServer(true);
-        zookeeperPort = zookeeper.getPort();
+        if (zookeeperPort == -1) {
+            zookeeper = new TestingServer(true);
+            zookeeperPort = zookeeper.getPort();
+        }else{
```

Comment:
Nit: please add space before and after the braces to keep the formatting consistent. 

 `} else {`